modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Iri.scala

```diff
@@ -960,17 +1004,32 @@ object Iri {
       * @param rest the remainder of the path (excluding the segment denoted by this)
       */
     final case class Segment private[rdf] (segment: String, rest: Path) extends Path {
-      def isEmpty: Boolean           = false
-      def isSlash: Boolean           = false
-      def isSegment: Boolean         = true
-      def asEmpty: Option[Empty]     = None
-      def asSlash: Option[Slash]     = None
-      def asSegment: Option[Segment] = Some(this)
-      def asString: String           = rest.asString + segment
-      def pctEncoded: String         = rest.pctEncoded + pctEncode(segment)
-      def startWithSlash: Boolean    = rest.startWithSlash
-      def prepend(other: Path): Path = (rest prepend other) + segment
-      def +(s: String): Path         = if (segment.isEmpty) this else Segment(segment + s, rest)
+      type Head = String
+
+      def isEmpty: Boolean               = false
+      def head: String                   = segment
+      def tail(skipSlash: Boolean): Path = if (skipSlash && rest.endsWithSlash) rest.tail(skipSlash) else rest
+      def isSlash: Boolean               = false
+      def isSegment: Boolean             = true
+      def asEmpty: Option[Empty]         = None
+      def asSlash: Option[Slash]         = None
+      def asSegment: Option[Segment]     = Some(this)
+      def asString: String               = rest.asString + segment
+      def pctEncoded: String             = rest.pctEncoded + pctEncode(segment)
+      def startWithSlash: Boolean        = rest.startWithSlash
+      def prepend(other: Path): Path     = (rest prepend other) + segment
+      def +(s: String): Path             = if (segment.isEmpty) this else Segment(segment + s, rest)
+      def toSegments: List[String] = {
```

Comment:
wouldn't it be simpler like so:
```
def toSegments = (segment :: rest.toSegments).reverse
```
... or use a vector? (because list concatenation is fairly slow)